hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadIncrementalRelation.scala

```diff
@@ -74,7 +74,8 @@ class MergeOnReadIncrementalRelation(val sqlContext: SQLContext,
     optParams.getOrElse(DataSourceReadOptions.END_INSTANTTIME.key, lastInstant.getTimestamp))
   log.debug(s"${commitsTimelineToReturn.getInstants.iterator().toList.map(f => f.toString).mkString(",")}")
   private val commitsToReturn = commitsTimelineToReturn.getInstants.iterator().toList
-  private val schemaUtil = new TableSchemaResolver(metaClient)
+  private val withOperationField: Boolean = new TableSchemaResolver(metaClient).hasOperationField
+  private val schemaUtil = new TableSchemaResolver(metaClient, withOperationField)
```

Comment:
We need to make things more clear here, `withOperationField` flag can actually be eliminated if we have assumption:
`always show the __hoodie_operation field when the schema contains that`. Then the resolve `withOperationField` can be made builtin in `TableSchemaResolver`.

To make sure the assumption, you should check all the use cases that pass `true` as `withOperationField` for `TableSchemaResolver`.